examples/bls381/src/wasm.rs

```diff
@@ -0,0 +1,182 @@
+// This file is part of Gear.
+
+// Copyright (C) Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+use ark_bls12_381::{Bls12_381, G1Affine, G2Affine};
+use ark_ec::pairing::Pairing;
+use gbuiltin_bls381::*;
+use gstd::{
+    codec::{Decode, Encode},
+    msg,
+    prelude::*,
+    ActorId,
+};
+use hex_literal::hex;
+
+type ArkScale<T> = ark_scale::ArkScale<T, { ark_scale::HOST_CALL }>;
+
+const BUILTIN_BLS381: ActorId = ActorId::new(hex!(
```

Comment:
Could we mb add it in gbuiltin crate mb with hint that it needs to be checked on-chain manually if you dont trust?

Reply:
As far as I remember we've discussed that already and  haven't found suitable solution. So leave it as is for the moment.